packages/store/src/internal/config-validator.ts

```diff
@@ -1,33 +1,43 @@
-import { Injectable } from '@angular/core';
+import { Injectable, isDevMode } from '@angular/core';
 
 import {
-  CONFIG_MESSAGES as MESSAGES,
-  VALIDATION_CODE as CODE
+  getIncorrectProductionMessage,
+  getIncorrectDevelopmentMessage
 } from '../configs/messages.config';
 import { NgxsConfig } from '../symbols';
 import { HostEnvironment } from '../host-environment/host-environment';
 
-@Injectable()
+@Injectable({ providedIn: 'root' })
 export class ConfigValidator {
   constructor(private _host: HostEnvironment, private _config: NgxsConfig) {}
 
-  private get isIncorrectProduction(): boolean {
-    return !this._host.isDevMode() && this._config.developmentMode;
-  }
-
-  private get isIncorrectDevelopment(): boolean {
-    return this._host.isDevMode() && !this._config.developmentMode;
-  }
-
-  public verifyDevMode(): void {
+  verifyDevMode(): void {
     if (this._host.isTestMode()) {
       return;
     }
 
-    if (this.isIncorrectProduction) {
-      console.warn(MESSAGES[CODE.INCORRECT_PRODUCTION]());
-    } else if (this.isIncorrectDevelopment) {
-      console.warn(MESSAGES[CODE.INCORRECT_DEVELOPMENT]());
+    if (typeof ngDevMode !== 'undefined') {
+      // Caretaker note: these conditions are placed here and not moved to separate
+      // methods or functions because they'll be tree-shaken away. Basically Terser will encounter
+      // the following expression:
+      // if (true) {
+      //   if (false && this._config.developmentMode) { ... }
+      //   else if (!false && this._config.developmentMode) { ... }
+      // }
+      // The `getIncorrectDevelopmentMessage()` will be tree-shaken away.
+      if (ngDevMode && !this._config.developmentMode) {
+        console.warn(getIncorrectDevelopmentMessage());
+      } else if (!ngDevMode && this._config.developmentMode) {
+        console.warn(getIncorrectProductionMessage());
+      }
+    } else {
+      const ngDevMode = isDevMode();
+      // If `ngDevMode` is undefined then it means we're using the View Engine compiler.
+      if (ngDevMode && !this._config.developmentMode) {
```

Comment:
do you think this parameter is now meaningless?

because now 

`config.developmentMode === ngDevMode`

Reply:
I mean, can now not force users to specify this parameter?  but just check for ngDevMode in the code and then you can pass this parameter deprecated

Reply:
I was also thinking if we still need this parameter or not. Basically it's used only to checker whether we have to freeze the state through `deepFreeze` or not. I'm wondering if we can remove the `ConfigValidator` completely and move all checks to `state-operations.ts`.

Reply:
@arturovt I agree with you on everything this